### PR TITLE
Add infra for deploy-gke CI workflows

### PR DESCRIPTION
**Pull request by JeromeJu**
This commit adds the infra variables for setting up the deploy-gke CI workflow inputs.

### CODE REVIEW

**Comment by JeromeJu** [comment]
One question I want to confirm with is that we do not essentially need to have the subnetwork set up as in https://github.com/google-github-actions/test-infra/blob/main/project_get-gke-credentials.tf#L108-L125?

**Comment by sethvargo** [comment]
> One question I want to confirm with is that we do not essentially need to have the subnetwork set up as in https://github.com/google-github-actions/test-infra/blob/main/project_get-gke-credentials.tf#L108-L125 IIUC (since that is to mock client network use case through a private cluster?).

correct

File: project_deploy-gke.tf [hunk 1]
```diff
@@ -28,6 +28,15 @@ module "deploy-gke" {
     "deployment",
   ], local.common_topics)
 
+  repo_variables = {
+    "IMAGE"           = google_container_cluster.deploy-gke.image
```
[the remark this answers is not in the document]
Comment (sethvargo) [reply project_deploy-gke.tf:32]: Yea - you'll need to create the cluster and set the IAM permissions.

[the remark this answers is not in the document]
Comment (JeromeJu) [reply project_deploy-gke.tf:32]: Thanks for confirming. 🙏 

I followed the internal onboarding doc and it seems I do not have access through the `graphi**it` GCP Project. I thought I would need to create a cluster under this [project_id](https://github.com/google-github-actions/test-infra/blob/4356c34c167336c7a4747c5b5923ecfa2fe3cbd6/main.tf#L89).

Is this the expected path to set up the cluster?

[the remark this answers is not in the document]
Comment (sethvargo) [reply project_deploy-gke.tf:32]: Hey @JeromeJu I can give you access, but you'll need me or Mike to actuate the infrastructure changes.

[the remark this answers is not in the document]
Comment (JeromeJu) [reply project_deploy-gke.tf:32]: Much appreciated  🙏 - I can also add this pointer into the internal onboarding instructions if this is the best practice

[the remark this answers is not in the document]
Comment (sethvargo) [reply project_deploy-gke.tf:32]: Thanks! Ping me on chat if you run into issues with the Terraform bits

[the remark this answers is not in the document]
Comment (JeromeJu) [reply project_deploy-gke.tf]: Thanks for the patience here @sethvargo  - I previously had some misunderstanding about the IAM permissions that I need to have (with GitHub, we actually don't need LatchKey set up such things). I've updated this PR PTAL

**Review by sethvargo — APPROVED** [review]
Seems reasonable - do you want me to try and apply these changes?

**Comment by JeromeJu** [comment]
> Seems reasonable - do you want me to try and apply these changes?

Thanks for your time! That would be much appreciated.

**Pull request closed without merging**